go/vt/vtgate/planbuilder/operators/SQL_builder.go

```diff
@@ -376,14 +376,33 @@ func buildQuery(op Operator, qb *queryBuilder) {
 	case *Update:
 		buildUpdate(op, qb)
 	case *Delete:
-		buildDML(op, qb)
+		buildDelete(op, qb)
 	case *Insert:
 		buildDML(op, qb)
 	default:
 		panic(vterrors.VT13001(fmt.Sprintf("unknown operator to convert to SQL: %T", op)))
 	}
 }
 
+func buildDelete(op *Delete, qb *queryBuilder) {
+	buildQuery(op.Source, qb)
+	qb.dmlOperator = op
+
+	sel, ok := qb.stmt.(*sqlparser.Select)
+	if !ok {
+		panic(vterrors.VT13001("expected a select here"))
+	}
```

Comment:
It would be nice to have a comment explaining why the Delete's source has to result in a select query in the buildQuery call. I understand why, but it took me a while too. Having a comment would be nice for future reference.